app_dart/lib/src/request_handlers/check_flaky_builders.dart

```diff
@@ -52,20 +52,44 @@ class DeflakeFlakyBuilders extends ApiRequestHandler<Body> {
     final BigqueryService bigquery = await config.createBigQueryService();
     final String ciContent = await gitHub.getFileContent(slug, kCiYamlPath);
     final List<_BuilderInfo> eligibleBuilders = await _getEligibleFlakyBuilders(gitHub, slug, content: ciContent);
-    String? testOwnerContent;
+    final List<BuilderStatistic> stagingBuilderStatisticList =
+        await bigquery.listBuilderStatistic(kBigQueryProjectId, bucket: 'staging');
+    final String testOwnerContent = await gitHub.getFileContent(slug, kTestOwnerPath);
     for (final _BuilderInfo info in eligibleBuilders) {
+      final BuilderType type = getTypeForBuilder(info.name, loadYaml(ciContent) as YamlMap);
+      final TestOwnership testOwnership = getTestOwnership(info.name!, type, testOwnerContent);
       final List<BuilderRecord> builderRecords =
           await bigquery.listRecentBuildRecordsForBuilder(kBigQueryProjectId, builder: info.name, limit: kRecordNumber);
       if (_shouldDeflake(builderRecords)) {
-        testOwnerContent ??= await gitHub.getFileContent(slug, kTestOwnerPath);
-        await _deflakyPullRequest(gitHub, slug, info: info, ciContent: ciContent, testOwnerContent: testOwnerContent);
+        await _deflakyPullRequest(gitHub, slug, info: info, ciContent: ciContent, testOwnership: testOwnership);
+      } else if (_shouldFileIssue(builderRecords, info)) {
+        final BuilderDetail builderDetail = BuilderDetail(
+          statistic: stagingBuilderStatisticList
+              .where((BuilderStatistic builderStatistic) => builderStatistic.name == info.name)
+              .single,
+          existingIssue: null,
+          existingPullRequest: null,
+          isMarkedFlaky: true,
+          type: type,
+          ownership: testOwnership,
+        );
+        await fileFlakyIssue(builderDetail: builderDetail, gitHub: gitHub, slug: slug, bringup: true);
       }
     }
     return Body.forJson(const <String, dynamic>{
       'Status': 'success',
     });
   }
 
+  /// A new issue should be filed for staging builders if
+  ///   1) there is any flake in recent runs
+  ///   2) there is no open flaky bug tracking the flake
+  bool _shouldFileIssue(List<BuilderRecord> builderRecords, _BuilderInfo info) {
+    final bool noExistingOpenIssue =
+        info.existingIssue == null || info.existingIssue != null && info.existingIssue!.isClosed;
+    return builderRecords.any((BuilderRecord record) => record.isFlaky) && noExistingOpenIssue;
```

Comment:
maybe flip the order to short-circuit the buildRecords check?

Reply:
Done.